keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -323,12 +323,23 @@ class ApplicationService(
   ): ArtifactVersionSummary {
     val artifactSupplier = artifactSuppliers.supporting(artifact.type)
     val publishedArtifact = artifact.toSpinnakerArtifact(version)
+    val artifactStatus = artifactSupplier.getReleaseStatus(publishedArtifact)
+    val buildMetadata = repository.getArtifactBuildMetadata(artifact.name, artifact.type, version, artifactStatus)
+    val gitMetadata = repository.getArtifactGitMetadata(artifact.name, artifact.type, version, artifactStatus)
     return ArtifactVersionSummary(
       version = version,
       environments = environments,
       displayName = artifactSupplier.getVersionDisplayName(publishedArtifact),
-      build = artifactSupplier.getBuildMetadata(publishedArtifact, artifact.versioningStrategy),
-      git = artifactSupplier.getGitMetadata(publishedArtifact, artifact.versioningStrategy)
+
+      // first attempt to fetch the artifact metadata from the DB, then fallback to the default if not found
```

Comment:
Ah, now I see why you kept those functions. In this case, I'd suggest renaming them so it's clear that they're not really going to retrieve additional data from anywhere, but instead just try to parse data out of the artifact. Maybe `ArtifactSupplier.parseBuildMetadata()` and `parseGitMetadata()` (or `extract` if you like that better)?

Reply:
changed to `getDefaultXXMetadata` -- let me know if it makes sense.

Reply:
I think `getDefaultXyzMetadata` doesn't really give any hints about what the function is expected to do, as "default" doesn't have any intrinsic meaning here, to the best of my knowledge. I'd prefer something like `extractXyzMetadata` as suggested earlier, but up to you.